conans/client/graph/graph_builder.py

```diff
@@ -304,17 +305,16 @@ def _config_node(node, down_ref, down_options):
             # Avoid extra time manipulating the sys.path for python
             with get_env_context_manager(conanfile, without_python=True):
                 if hasattr(conanfile, "config"):
-                    if not ref:
-                        conanfile.output.warn("config() has been deprecated."
-                                              " Use config_options and configure")
+                    conan_v2_behavior("config() has been deprecated. Use config_options() and configure()",
```

Comment:
It wasn't my intention, I'll keep the old behavior to avoid too much noise.

Reply:
Nope. I need to change this, otherwise the message is not shown and this test fails:

```python
    def test_config_method(self):
        t = self.get_client()
        conanfile = textwrap.dedent("""
            from conans import ConanFile

            class Recipe(ConanFile):
                def config(self):
                    pass
        """)
        t.save({'conanfile.py': conanfile})
        t.run('create . name/version@', assert_error=True)
        self.assertIn("Conan v2 incompatible: config() has been deprecated. Use config_options() and configure()",
                      t.out)
```